### DISCUSSION TITLE
Using CLI to run fabric with links or files

### DISCUSSION THREAD

**Discussion by alexindelicato-github** [danielmiessler/Fabric#345] [Q&A] [answered]
Hello, I am looking for some help here :| . I can use the fabric GUI to provide links and help get summary of articles, but I cannot quite figure out what the CLI prompt should be. I have tried several formats including pbpaste (running WSL). 

`echo 'https://thehackernews.com/2024/04/palo-alto-networks-releases-urgent.html' | fabric --pattern summarize`

Response:

I'm sorry, but I can't access or retrieve content from external links, including websites like The Hacker News. However, if you provide me with details or a summary of the article from Palo Alto Networks regarding an urgent update, I can help create a summary based on the information you share.

This works of course if I copy and paste text, but looking for a simpler way to have written articles or documents summarize. Similar how would one do this with a PDF (I understand after a certain size, files may need to go through the RAG process and chunked). 

Thanks in advanced

**Comment by xssdoctor** [comment, marked as answer]
You can't echo a web site. You need to copy the text of the site and then paste it or use pbpaste or xclip

**Reply by alexindelicato-github** [reply]
Thank you for confirming.

**Reply by SumoSumir** [reply]
I have tried `curl https://thehackernews.com/2024/04/palo-alto-networks-releases-urgent.html | fabric --pattern summarize` and it works, but it's not a definitive solution for pages that need javascript to load content (using claude btw)

**Reply by patonw** [reply]
Yeah, curl is the way to go, but I think dynamic content is less of a problem on pages that you'd want to summarize like articles and such. The main issue would probably be all the extraneous navigation and formatting tag elements adding noise to the input. You can filter that out using a tool like htmlq: https://github.com/mgdm/htmlq

Edit: for that example, you'd use `htmlq '#articlebody' --text`

**Reply by alexindelicato-github** [reply]
Thank you both, this is super helpful.

**Comment by BunkerBuster73** [comment]
hi, sorry may be i not understand what you want to do but this is not working ?

fabric -t "https://thehackernews.com/2024/04/palo-alto-networks-releases-urgent.html" -p summarize

**Reply by michaellerickson** [reply]
This will not work: 

fabric -t "https://thehackernews.com/2024/04/palo-alto-networks-releases-urgent.html" -p summarize

For now, what you will need to do is visit the website, copy the contents and use pbpaste OR you could save the file and call it from the file with a pipe, for example: cat sitecontents.txt | fabric -C
